src/buildSummary.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 The MathWorks, Inc.
+import * as core from "@actions/core";
+import { join } from 'path';
+import { readFileSync } from 'fs';
+
+export interface Task {
+  name: string;
+  description: string;
+  failed: boolean;
+  skipped: boolean;
+  duration: string;
+}
+
+
+export function getBuildSummaryTable(tasks: Task[]): string[][] {
+  const header: string[] = ['MATLAB Build Task', 'Status', 'Description', 'Duration (HH:MM:SS)'];
+  let taskSummaryTableRows: string[][] = [header];
+
+  tasks.forEach((task, index) => {
+    let taskDetails: string[] = [];
+    taskDetails.push(task.name);
+    if (task.failed) {
+      taskDetails.push('🔴 FAILED');
+    } else if (task.skipped) {
+      taskDetails.push('🔵 SKIPPED');
+    } else {
+      taskDetails.push('🟢 SUCCESS');
+    }
+    taskDetails.push(task.description);
+    taskDetails.push(task.duration);
+
+    taskSummaryTableRows.push(taskDetails);
+  });
+
+  return taskSummaryTableRows;
+}
+
+export function writeSummary(taskSummaryTableRows: string[][]) {
+  core.summary
+    .addTable(taskSummaryTableRows)
+    .write();
+}
+
+export function processAndDisplayBuildSummary() {
+  const runId = process.env.GITHUB_RUN_ID;
+  const runnerTemp = process.env.RUNNER_TEMP;
+  let filePath: string;
+
+  if (!runId) {
+    filePath = join(runnerTemp as string, `buildSummary_.json`);
```

Comment:
What does it mean for the `runId` to be undefinded? I feel like we might have bigger issues than just the naming of the summary file if that's the case.

Also, could we change `buildSummary_.json` to `buildSummary.json`? I think the trailing underscore looks a bit strange.

Reply:
Agree ! Done.

Reply:
Its just to make sure ` GITHUB_RUN_ID` is available. more than naming I think its to maintain uniqness for each run and avoid naming conflicts. Althought in ost cases `GITHUB_RUN_ID` is set its just fail safe mechanism just in case the env variable is not set for any reasons.